frontend/test/pytest/test_preprocess.py

```diff
@@ -65,61 +77,49 @@ def preprocess(self, execution_config: ExecutionConfig = DefaultExecutionConfig)
 
 
-class DummyDeviceCounts(Device):
-    """A dummy device from the device API without wires."""
-
-    config = get_lib_path("runtime", "RUNTIME_LIB_DIR") + "/backend/dummy_device.toml"
-
-    def __init__(self, wires, shots=1024):
-        super().__init__(wires=wires, shots=shots)
-
-    @staticmethod
-    def get_c_interface():
-        """Returns a tuple consisting of the device name, and
-        the location to the shared object with the C/C++ device implementation.
-        """
+class OtherHadamard(qml.Hadamard):
+    """A version of the Hadamard operator that won't be recognized by the QJit device, and will
+    need to be decomposed"""
 
-        system_extension = ".dylib" if platform.system() == "Darwin" else ".so"
-        lib_path = get_lib_path("runtime", "RUNTIME_LIB_DIR") + "/librtd_dummy" + system_extension
-        return "dummy.remote", lib_path
+    @property
+    def name(self):
+        """name"""
+        return "UnknownOp"
 
-    def execute(self, circuits, execution_config):
-        """Execution."""
-        return circuits, execution_config
 
-    def __enter__(self, *args, **kwargs):
-        dummy_toml = self.config
-        with open(dummy_toml, mode="r", encoding="UTF-8") as f:
-            toml_contents = f.readlines()
+class OtherIsingXX(qml.IsingXX):
+    """A version of the IsingXX operator that won't be recognized by the QJit device, and will
+    need to be decomposed"""
 
-        updated_toml_contents = []
-        for line in toml_contents:
-            if "Expval" in line:
-                continue
-            if "Var" in line:
-                continue
-            if "Probs" in line:
-                continue
-            if "Sample" in line:
-                continue
+    @property
+    def name(self):
+        """name"""
+        return "UnknownOp"
 
-            updated_toml_contents.append(line)
 
-        self.toml_file = tempfile.NamedTemporaryFile(mode="w", delete=False)
-        self.toml_file.writelines(updated_toml_contents)
-        self.toml_file.close()  # close for now without deleting
+class OtherRX(qml.RX):
+    """A version of the RX operator that won't be recognized by the QJit device, and will need to
+    be decomposed"""
 
-        self.config = self.toml_file.name
-        return self
+    @property
+    def name(self):
+        """Name of the operator is UnknownOp"""
+        return "UnknownOp"
 
-    def __exit__(self, *args, **kwargs):
-        os.unlink(self.toml_file.name)
-        self.config = None
+    def decomposition(self):
+        """decomposes to normal RX"""
+        return [qml.RX(*self.parameters, self.wires)]
 
 
 class TestPreprocess:
     """Test the preprocessing transforms implemented in Catalyst."""
 
+    @pytest.mark.skipif(
+        not pathlib.Path(
+            get_lib_path("runtime", "RUNTIME_LIB_DIR") + "/libdummy_device.so"
+        ).is_file(),
+        reason="lib_dummydevice.so was not found.",
+    )
```

Comment:
I think this PR is great, but I am just a bit worried that these tests might not be run on macOS. Do you know if that is the case?

Reply:
This has been resolved offline. No changes required.